# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RenderD3D11TextureHostOGL.cpp
# Commit: 0067bb1fb8e4
# Full Hash: 0067bb1fb8e45078f4383c9e371dd1c7eafbffbe
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2020-07-29 09:49:32
# Regressor Bug: 1652972
# File Overlap Count: 1
# Description:
#   Bug 1652972 - Mark EGL context as lost on Shutdown() r=aosmond
#   
#   This is meant to save us in cases where the message loop in GPU process
#   receives commands related to resources that point to the old EGL context
#   that was just shut down. Since the symbols are erased, we'd end up with
# ==============================================================================

diff -r bd33c30ba4b5 -r 0067bb1fb8e4 gfx/webrender_bindings/RenderD3D11TextureHostOGL.cpp
--- a/gfx/webrender_bindings/RenderD3D11TextureHostOGL.cpp	Tue Jul 28 22:26:21 2020 +0000
+++ b/gfx/webrender_bindings/RenderD3D11TextureHostOGL.cpp	Wed Jul 29 02:33:54 2020 +0000
@@ -260,24 +260,24 @@
 
   if (mGL->MakeCurrent()) {
     mGL->fDeleteTextures(2, mTextureHandle);
+    const auto& gle = gl::GLContextEGL::Cast(mGL);
+    const auto& egl = gle->mEgl;
+    if (mSurface) {
+      egl->fDestroySurface(egl->Display(), mSurface);
+    }
+    if (mStream) {
+      egl->fDestroyStreamKHR(egl->Display(), mStream);
+    }
   }
+
   for (int i = 0; i < 2; ++i) {
     mTextureHandle[i] = 0;
   }
 
-  const auto& gle = gl::GLContextEGL::Cast(mGL);
-  const auto& egl = gle->mEgl;
-  if (mSurface) {
-    egl->fDestroySurface(egl->Display(), mSurface);
-    mSurface = 0;
-  }
-  if (mStream) {
-    egl->fDestroyStreamKHR(egl->Display(), mStream);
-    mStream = 0;
-  }
-
   mTexture = nullptr;
   mKeyedMutex = nullptr;
+  mSurface = 0;
+  mStream = 0;
 }
 
 GLuint RenderDXGITextureHostOGL::GetGLHandle(uint8_t aChannelIndex) const {
